crowbar_framework/app/models/service_object.rb

```diff
@@ -1513,15 +1489,8 @@ def add_role_to_instance_and_node(barclamp, instance, name, prop, role, newrole)
     end
 
     save_it = false
-    unless node.role?(newrole)
-      node.add_to_run_list(newrole, local_chef_order, role_map[newrole])
-      save_it = true
-    end
-
-    unless node.role?("#{barclamp}-config-#{instance}")
-      node.add_to_run_list("#{barclamp}-config-#{instance}", local_chef_order)
-      save_it = true
-    end
+    save_it = node.add_to_run_list(newrole, local_chef_order) || save_it
```

Comment:
Here there's an argument missing (`role_map[newrole]`) compared to previous version of `add_to_run_list` call. 

When I add it, my manual test seems to succeed and add the network data correctly.